src/vpart_range.h

```diff
@@ -101,14 +101,16 @@ class generic_vehicle_part_range
 {
     private:
         std::reference_wrapper<::vehicle> vehicle_;
+        bool include_fake_;
 
     public:
-        generic_vehicle_part_range( ::vehicle &v ) : vehicle_( v ) { }
+        generic_vehicle_part_range( ::vehicle &v, bool incl_fake = false ) : vehicle_( v ),
+            include_fake_( incl_fake ) { }
 
         // Templated because see top of file.
         template<typename T = ::vehicle>
         size_t part_count() const {
-            return static_cast<const T &>( vehicle_.get() ).parts.size();
+            return static_cast<const T &>( vehicle_.get() ).num_parts_incl_fake();
```

Comment:
Shouldn't this depend on the value of `include_fake_`? When it's true, the fake parts are including in the count, otherwise they are excluded?